pkg/controllers/resources/runtimeclasses/syncer.go

```diff
@@ -60,6 +67,12 @@ func (i *runtimeClassSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *
 }
 
 func (i *runtimeClassSyncer) Sync(ctx *synccontext.SyncContext, event *synccontext.SyncEvent[*nodev1.RuntimeClass]) (_ ctrl.Result, retErr error) {
+	if !ctx.Config.Sync.FromHost.RuntimeClasses.Selector.Matches(event.Host) {
+		i.EventRecorder().Eventf(event.Virtual, "Warning", "SyncWarning", "did not sync runtime class %q because it does not match the selector under 'sync.fromHost.runtimeClasses.selector'", event.Host.Name)
+		i.EventRecorder().Eventf(event.Host, "Warning", "SyncWarning", "did not sync runtime class %q because it does not match the selector under 'sync.fromHost.runtimeClasses.selector'", event.Host.Name)
```

Comment:
We cannot use event.Host here as the event recorder only works for the virtual object